python/cudf/cudf/core/udf/_nrt_cuda.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
```

Comment:
Are we eventually hoping for this whole module to go away and just be a part of Numba?

Reply:
Eventually I would like to see that but there's a reasonable gap between "enough functionality for cuDF's purposes" and "the complete Numba runtime functionality" that would be needed for the CUDA target to support it in general. I see this as a step along the way to adding that support.

Reply:
A lot of this can be upstreamed eventually yes. With NRT enabled, our cuda target contexts would already have `enable_nrt=True` and presumably the extra PTX we're feeding numba's linker here would be present for all cuda kernel launches.

Reply:
The extra PTX would at least be added when needed. I think if this logic were in Numba it would be much easier to add it exactly when it's required, rather than adding it to every link regardless.